### PR TITLE
V2 Prerelease - Phase 1

### PR DESCRIPTION
**Pull request by z3dev**
These changes create the MAJOR prerelease for V2. All package VERSION numbers have been changed to 2.0.0-alpha.0, which designates that **packages are in prerelease mode**.

Futher changes to package contents will increment the last digit of the VERSION, **not** the VERSION of the package. This allows the packages to be published (next step) with the same VERSION as fixes, etc are made.

A few changes to the lerna configuration were made.
- desktop / gcode-deserializer / scad-deserializer packages are now 'priviate' to prevent new versions / publishing
- vtree dev dependencies were corrected

Note: There is a new version of web/dist/jscad-full.min.js, which is a real V2 release.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

Tests also performed with WEB UI, both npm run dev and demo.html.

### CODE REVIEW

**Comment by z3dev** [comment]
I think now is the best time to validate the CHANGELOG for each of the packages. These seem incomplete and sometimes misleading.

**Comment by z3dev** [comment]
@SimonClark sorry to request a review, but i have integrated all the fixes and feats into the CHANGELOG files. this brings and end to PHASE 1, and the next increment should be far easier and automated.

Getting closer to a real release!

**Comment by z3dev** [comment]
[changes.txt](https://github.com/jscad/OpenJSCAD.org/files/5079876/changes.txt)

A list of all commits on V2 branch.

File: packages/io/svg-serializer/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,23 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+# 2.0.0-alpha.0 (2020-08-13)
+
+
+### Bug Fixes
+
+* **color:** color is now immutable & returns new geometries ([#566](https://github.com/jscad/OpenJSCAD.org/issues/566)) ([916824c](https://github.com/jscad/OpenJSCAD.org/commit/916824c63a0bf8896d1b6f85a82b129a013bec58))
```
Comment (SimonClark) [inline packages/io/svg-serializer/CHANGELOG.md:11]: "colors" :)

File: packages/modeling/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,50 @@
...
+* **primitives:** enhance primitives settings & defaults
+* **geometries:** added color support to to/fromCompactBinary functions
+* **maths:** updated vec2 and vec3 to use the same logic as glmatrix
+* **maths:** changed line2/mat4/vec2/vec3/vec4 to create arrays of Numbers
+* **measurements:** enhanced measurements to cache values as object attributes
+* **utils:** exposed additional user utility functions
+* **geometries:** corrected order of transforms in geom2, geom3, and path2
+
+
+### Features
```
Comment (SimonClark) [inline packages/modeling/CHANGELOG.md:27]: I guess I should add the curves stuff and the bugfixes to this as I commit them

**Comment by SimonClark** [comment]
Did you want me to merge your PRs when I approve them?

File: packages/modeling/CHANGELOG.md (hunk 2 above)
Comment (z3dev) [reply packages/modeling/CHANGELOG.md:27]: curves is mentioned in the Features, although not very apparent.

FYI...
- each Fix requires a patch bump in version... 2.0.0 to 2.0.1
- each Feature requires a minor bump in version... 2.0.0 to 2.1.0

File: packages/io/svg-serializer/CHANGELOG.md (hunk 1 above)
Comment (z3dev) [reply packages/io/svg-serializer/CHANGELOG.md]: done

**Comment by z3dev** [comment]
> Did you want me to merge your PRs when I approve them?

Sure. Just make sure...
- all questions have been answered
- all requested changes have been completed (i think this requires a review again anyway)
- all CI checks have passed (this is required to merge anyway)

**Pull request merged**